File: php/tests/smoke.php
<?php
require __DIR__ . '/../vendor/autoload.php';

use AriaBridge\BridgeConfig;
use AriaBridge\Client;

$url = getenv('ARIA_BRIDGE_URL') ?: 'ws://localhost:9877';
$secret = getenv('ARIA_BRIDGE_SECRET') ?: 'dev-secret';

$client = new Client(new BridgeConfig($url, $secret, 'php-test'));
$client->sendConsole('smoke test console');
$client->sendError('smoke test error');

if (!function_exists('pcntl_fork')) {
  echo "pcntl not available; skipping php smoke\n";
  exit(0);
}

$pid = pcntl_fork();
if ($pid === -1) {
  fwrite(STDERR, "fork failed\n");
  exit(1);
}

if ($pid === 0) {
  pcntl_async_signals(true);
  pcntl_signal(SIGALRM, function () use ($client) {
    $client->stop();
  });
  pcntl_alarm(4);
  $client->start();
  exit(0);
}

pcntl_waitpid($pid, $status);
echo "php smoke ok\n";

File: ruby/tests/smoke.rb
require_relative '../lib/aria_bridge_client'

url = ENV.fetch('ARIA_BRIDGE_URL', 'ws://localhost:9877')
secret = ENV.fetch('ARIA_BRIDGE_SECRET', 'dev-secret')

client = AriaBridge::Client.new(
  url: url,
  secret: secret,
  project_id: 'ruby-test',
  heartbeat_interval: 0.1,
  heartbeat_timeout: 0.2,
  backoff_initial: 0.05,
  backoff_max: 0.2
)
client.start
client.send_console('ruby smoke', level: 'info')
client.send_error('ruby error')
sleep 0.4
client.stop
puts 'ruby smoke ok'
